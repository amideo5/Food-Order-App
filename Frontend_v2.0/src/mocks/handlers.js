// src/mocks/handlers.js
import { rest } from "msw";

export const handlers = [
  rest.post("http://localhost:8080/users/signin", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Successful")
    );
  }),

  rest.post("http://localhost:8080/users/createUser", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Registered")
    );
  }),

  rest.get("http://localhost:8080/users/name/*", async (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.body("Aryan")
    );
  }),

  rest.get("http://localhost:8080/orders/getOrder/*", (req, res, ctx) => {
    return res(
      // Respond with a 200 status code
      ctx.status(200),
      ctx.json({
        orderName: "Aryan",
        items: {
          orderItems: {
            rajma: "2",
            chicken: "3",
          },
        },
        cost: "2000",
      })
    );
  }),

  rest.put("http://localhost:8080/users/updateUser/*", (req, res, ctx) => {
    return res(ctx.status(200), ctx.body("User Updated"));
  }),
  
];
