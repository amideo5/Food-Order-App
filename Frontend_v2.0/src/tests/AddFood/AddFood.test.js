import React from 'react';
import { getAllByText, render, screen } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import { server } from '../../mocks/server';
import { rest } from "msw";
import axios from 'axios';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { fireEvent, waitFor } from "@testing-library/react";
import Add from '../../Components/AddFood';


describe('Add component', () => {

    test("Snapshot testing",()=>{
        const {container} = render(<Add/>);

        expect(container).toMatchSnapshot();
    });
  
  
  
    it('updates the food name input', () => {
      render(<Add />);
      const foodNameInput = screen.getByPlaceholderText('Name');
      expect(foodNameInput).toBeInTheDocument();
      fireEvent.change(foodNameInput, 'New Food Name');
      expect(foodNameInput).toHaveValue('');
    });
  
    it('updates the food price input', () => {
      render(<Add />);
      const foodPriceInput = screen.getByPlaceholderText('Price');
      expect(foodPriceInput).toBeInTheDocument();
      fireEvent.change(foodPriceInput, '15');
      expect(foodPriceInput).toHaveValue('');
    });
  
    it('updates the food category input', () => {
      render(<Add />);
      const foodCategoryInput = screen.getByPlaceholderText('Category');
      expect(foodCategoryInput).toBeInTheDocument();
      fireEvent.change(foodCategoryInput, 'New Food Category');
      expect(foodCategoryInput).toHaveValue('');
    });


    test("should update food and redirect to /add page", async () => {

      server.resetHandlers(
          rest.post("http://localhost:8080/foods/createFood", (req, res, ctx) => {
            return res(
              // Respond with a 200 status code
              ctx.status(200),
              ctx.body("Food Created")
            );
          })
        );

          const props = {
              history: { push: jest.fn() },
            };

            const mockAlert = jest.fn();
            global.alert = mockAlert;
          const { getByPlaceholderText, getByText } = render(<Add {...props} />);

          fireEvent.change(getByPlaceholderText("Name"), {
            target: { value: "Name" },
          });
          fireEvent.change(getByPlaceholderText("Price"), {
              target: { value: "10" },
            });
          fireEvent.change(getByPlaceholderText("Category"), {
            target: { value: "Starter" },
          });
          const but = screen.getAllByText("Register Food");
          fireEvent.click(but[1]);

          await waitFor(() =>
          expect(mockAlert).toBeCalled());
        });

    
    
  });
  