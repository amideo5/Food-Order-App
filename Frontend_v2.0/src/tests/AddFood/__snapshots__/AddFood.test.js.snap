// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add component Snapshot testing 1`] = `
<div>
  <div
    class="page"
  >
    <nav
      class="m-0 py-3 navbar navbar-expand-lg navbar-dark bg-primary"
    >
      <div
        class="container-fluid"
      >
        <a
          class="navbar-brand"
          href="/welcome"
        >
          NEX Food Ordering
        </a>
        <button
          aria-controls="responsive-navbar-nav"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="responsive-navbar-nav"
        >
          <div
            class="me-auto navbar-nav"
          >
            <a
              class="nav-link"
              href="#"
              role="button"
              tabindex="0"
            />
            <a
              class="nav-link"
              data-rr-ui-event-key="/welcomea"
              href="/welcomea"
            >
              Admin Home
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/add"
              href="/add"
            >
              Add Food
            </a>
            <a
              class="nav-link"
              data-rr-ui-event-key="/update"
              href="/update"
            >
              Update Food
            </a>
          </div>
          <div
            class="navbar-nav"
          >
            <a
              class="nav-link"
              data-rr-ui-event-key="/admin"
              href="/admin"
            >
              Logout
            </a>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="form-container"
    >
      <div
        class="form-box-signup"
      >
        <div
          class="header-form"
        >
          <h4
            class="text-primary text-center"
          >
            <i
              class="fa fa-user-circle"
              style="font-size: 110px;"
            />
          </h4>
          <div
            class="image"
          />
        </div>
        <h2>
          Register Food
        </h2>
        <br />
        <div
          class="body-form"
        >
          <form
            class="m-3"
          >
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              />
              <input
                class="form-control"
                name="foodName"
                placeholder="Name"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              />
              <input
                class="form-control"
                name="foodPrice"
                placeholder="Price"
                required=""
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              />
              <input
                class="form-control"
                name="foodCategory"
                placeholder="Category"
                required=""
                type="text"
                value=""
              />
            </div>
            <br />
            <button
              class="btn btn-secondary btn-block"
              type="submit"
            >
              Register Food
            </button>
            <br />
            <br />
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
